chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -251,8 +257,8 @@ if test $KEEP_EXISTING_HOSTNAME -ne 1; then
   HOSTNAME="d${MAC//:/-}.${DOMAIN}"
 else
   HOSTNAME=$(cat /etc/HOSTNAME)
-  if ! echo "$HOSTNAME" | grep -q -o "$DOMAIN$"; then
-    echo "The fully qualified domain name did not contain the cloud domain."
+  if ! echo "$HOSTNAME" | grep -q -o "\.$DOMAIN$" || echo "${HOSTNAME%%.$DOMAIN}" | grep -q -o "\."; then
```

Comment:
(stupid question: why do we need `-o`?)

I'm not sure the error string below makes sense for the second test (`echo "${HOSTNAME%%.$DOMAIN}" | grep -q -o "\."`); I think I'd just do a `elif` with that test and a different string.